packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -242,6 +245,34 @@ class DropdownMenu<T> extends StatefulWidget {
   /// If [showTrailingIcon] is false, the trailing icon will not be shown.
   final Widget? trailingIcon;
 
+  /// Defines the extra padding around [leadingIcon].
+  ///
+  /// This padding is not the property of leading icon widget,
+  /// it is an extra layer of padding between leading icon widget and the text input field.
+  ///
+  /// Defaults to EdgeInsets.all(8.0).
+  final EdgeInsetsGeometry? leadingIconPadding;
```

Comment:
Instead of adding new APIs that only change the leading and trailing icon paddings. I think maybe we should add "trailingButton" and "leadingButton" so we can set something like:
```dart
final Widget trailingButton = widget.showTrailingIcon
            ? **widget.trailingButton** ?? Padding(...)
            :  ...
```

Similar for leading button. I think maybe this can be better for leading and trailing button customization. The existing `trailingIcon` and `leadingIcon` look a bit confusing and limit the customizability. We can consider to deprecate them.

Reply:
FYI, this is something I'm currently working on in the context of https://github.com/flutter/flutter/issues/174096 and https://github.com/flutter/flutter/issues/175847 (it will also help for other issues).